### PR TITLE
Feat: Encrypt pool passphrase with operator password, add adm-pool CLI and /api/health

### LINKED ISSUE #12: [Enhancement] Encrypt the pool passphrase with an operator password + adm-pool control CLI + /api/health
## Summary

The pool's delegate passphrase is currently stored in plain text in the config file (`passPhrase` in `config.jsonc`). This issue tracks adding support for **operator-password-encrypted passphrases** so that a leaked config file does not immediately expose the forging/payout key, plus an **`adm-pool` control CLI** to encrypt/unlock/inspect the running pool and an **`/api/health`** endpoint for external monitoring (e.g. Zabbix).

This mirrors the node-side work in [Adamant-im/adamant#247](https://github.com/Adamant-im/adamant/issues/247) (see its comments for the lock/unlock flow and CLI sketch), adapted to the pool's single-passphrase, payout-and-notify model.

## Motivation

The pool runs as a long-lived service (usually under pm2) on a host the operator may not fully control. The passphrase in `config.jsonc` is the delegate's secret — with it an attacker can move the pool's funds and impersonate the delegate. If the config leaks (backup, wrong file permissions, supply-chain tooling), the secret is exposed at rest.

Encrypting the passphrase with a single operator password raises the bar for credential theft while keeping operations simple. Backward compatibility with a plain passphrase is preserved for existing deployments.

Separately, operators need a machine-readable health endpoint so monitoring (Zabbix, Uptime Kuma, etc.) can alert on pool problems without scraping logs or the dashboard.

### What the passphrase is used for

In the pool the passphrase is only needed for **outgoing signed actions**:

- sending payout transactions to voters — `server/src/modules/pay_out.js`
- sending ADAMANT notification messages — `server/src/helpers/notify.js`

Everything else (block sync, reward calculation, the public JSON API at `server/src/api/index.js`, and the web dashboard) is read-only and needs no secret. This means the pool can run fully **except payouts/ADM notifications** while the passphrase is locked.

### Locked vs. unlocked

When an encrypted passphrase is detected, the pool boots into a **LOCKED** state:

```
pm2 start/restart adamant-pool
        ↓
pool boots normally
        ↓
config is loaded
        ↓
encrypted passphrase detected
        ↓
pool marks payouts/notifications as LOCKED
        ↓
block sync, reward calculation, web UI, public API all work normally
        ↓
operator runs `adm-pool unlock`
        ↓
passphrase decrypted in memory, keypair derived
        ↓
payouts & ADM notifications enabled; any payout due during the lock is processed
```

While locked:
- The scheduled payout run does **not** sign or broadcast — the period is computed and the pending amount accumulates, but the payout is **deferred** until unlock (no rewards are lost; the next eligible run picks them up).
- ADAMANT message notifications are skipped; non-passphrase notification channels (Slack/Discord, if configured) still work and should warn that the pool is locked.
- `/api/health` reports the locked state (see below).

### Startup behavior: interactive vs. service

- **Foreground / TTY (`npm start` in a terminal):** if the passphrase is encrypted, prompt for the operator password interactively and block startup until it is entered (or the operator chooses to start locked).
- **Service / no TTY (pm2, systemd, Docker):** never block boot on stdin. Start in the LOCKED state and keep running (sync + API + UI). The operator unlocks out-of-band with `adm-pool unlock`, which talks to the running process over a local control socket.

### Encryption design sketch

- Use a well-established symmetric scheme (e.g. **AES-256-GCM**) with a key derived from the operator password via a memory-hard KDF (e.g. **scrypt** or **Argon2id**).
- Store the encrypted value as a **self-describing string** in the config (a prefixed base64 blob containing algorithm id, KDF params, salt, IV, and ciphertext) so plain vs. encrypted detection is **deterministic**, not heuristic (no guessing from length/charset).
- The password lives in process memory only for the decryption step; it is never written to disk, logs, or any storage. On restart it must be supplied again.

### CLI: `adm-pool`

A small control CLI that ships with the pool:

```
adm-pool encrypt    # prompt for plain passphrase + operator password, print the encrypted config string
adm-pool unlock     # prompt for the operator password, unlock the running pool
adm-pool status     # show the running pool's status (also available unauthenticated via /api/health)
adm-pool lock       # (optional) re-lock: clear the decrypted keypair from memory
```

Example sessions:

```
$ adm-pool encrypt
# Passphrase: ********
# Operator password: ********
# Confirm operator password: ********
# Encrypted passphrase (put this in config.jsonc "passPhrase"):
# enc:v1:argon2id:...base64...

$ adm-pool status
# Pool:      MyDelegate (U1234...)
# Payouts:   locked
# Passphrase: encrypted, not loaded
# Node sync: synced
# Next payout run: 2026-07-01T00:00:00Z (deferred — locked)

$ adm-pool unlock
# Operator password: ********
# Pool unlocked. Payouts and ADM notifications enabled.

$ adm-pool lock
# Pool locked. Decrypted keypair cleared from memory.
```

`encrypt` runs standalone (no running pool needed). `unlock` / `lock` / `status` talk to the running pool over a local **control socket**, analogous to the node's `adamant-control.sock` — e.g. a Unix domain socket under the pool data dir (path configurable, default permissions `0600`, owner-only).

### `/api/health` endpoint

Add a GET endpoint to the existing read-only API (`server/src/api/index.js`) for external monitoring:

```
GET /api/health
200 OK
{
  "status": "ok",                 // ok | degraded | error
  "version": "3.1.0",
  "uptime": 12345,
  "node": { "synced": true, "height": 1234567 },
  "payouts": "unlocked",          // unlocked | locked
  "passphrase": "encrypted",      // encrypted | plain
  "lastPayoutTimestamp": 1750000000000,
  "nextPayoutTimestamp": 1750086400000
}
```

- Returns `200` when healthy. Locked-but-otherwise-fine could stay `200` with `payouts: "locked"`, or `503`/`degraded` if the operator wants monitoring to alert on the locked state — make this explicit and documented so Zabbix templates are stable.
- The endpoint exposes **no secrets** and accepts **no input** — consistent with the existing GET-only, secret-free API contract.

### Backward compatibility

Until a separate deprecation notice is issued, the pool continues to accept a **plain-text passphrase** in the config with no migration required (for compatibility; not recommended). Detection is deterministic via the encrypted blob's self-describing prefix.

### Security constraints

- Never log the operator password, the decrypted passphrase, or any intermediate key material.
- Choose adequate KDF parameters to resist trivial brute force.
- Control socket is local-only, owner-readable (`0600`); no network-exposed unlock.
- No weakening of existing validation; `/api/health` adds no unvalidated input.

## Alternatives

- **Secrets manager / Vault integration** — more powerful but a hard external dependency, at odds with the lightweight, self-hosted pool.
- **OS keychain** — platform-specific; complicates headless/Docker.
- **Env var only (no encryption)** — still exposes the secret in process tables and shell history; does not solve the at-rest config leak.

## Proposed technical implementation

- **Encryption helper** (e.g. `server/src/helpers/crypto/passphrase.js`): `encrypt(plain, password)` / `decrypt(blob, password)` + a deterministic `isEncrypted(value)`. Unit-tested for round-trip and for plain/encrypted detection.
- **Config loading** (`server/src/helpers/config/reader.js`): detect encrypted `passPhrase`; if encrypted, do **not** derive the keypair at load time — defer to unlock. Update `schema.js` to allow the encrypted format. Keep the plain path working.
- **Lock state + control socket** in the server bootstrap (`server/src/app.js`): hold payout/notify behind a "secret available" gate; expose unlock/lock/status over the socket.
- **Gate the signing paths** in `pay_out.js` and `notify.js` so they no-op (and defer payouts) while locked.
- **`adm-pool` CLI** (e.g. `server/bin/adm-pool.js`, wired as a `bin` entry or `npm run` script): `encrypt`, `unlock`, `lock`, `status`.
- **`/api/health`** route in `server/src/api/index.js`.
- **Docs**: `README.md` + `config.default.jsonc` comments for the encrypt/unlock workflow, pm2 locked-boot behavior, the control socket path, and a Zabbix-friendly description of `/api/health`.

## References

- Upstream node issue: [Adamant-im/adamant#247 — Encrypt delegate passphrases with operator password](https://github.com/Adamant-im/adamant/issues/247) (see comments for the boot/unlock flow, the `adm-node forging {encrypt,unlock,lock,status}` CLI sketch, and the control-socket approach).

### PR DESCRIPTION
**Pull request by massivedev0**
## Description

Adds optional operator-password encryption for the pool passphrase, an `adm-pool` control CLI to manage it, and an `/api/health` endpoint for external monitoring. A plain passphrase still works unchanged, so the change is backward compatible.

Passphrase encryption:

- encrypts the passphrase with AES-256-GCM and a key derived from the operator password via scrypt, using only Node's built-in `crypto` (no new dependencies)
- stores the result as a self-describing `admpool-enc-v1.` blob (KDF params, salt, IV, auth tag, ciphertext) so plain vs encrypted detection is deterministic, never a heuristic
- embeds the non-secret public key and address in the blob so the pool can run read-only work while locked, and verifies the decrypted passphrase against that public key on unlock
- the operator password and decrypted passphrase live in memory only and are never written to disk or logs

Lock state and startup:

- when the passphrase is encrypted the pool boots LOCKED: block sync, the web dashboard, and the public API all work, but payouts and ADM notifications are paused until unlock
- in a terminal the pool prompts for the operator password on startup with masked input; under pm2/systemd (no TTY) it keeps running locked and waits for `adm-pool unlock`
- delegate/network startup is held until the interactive unlock is resolved, so it cannot exit the process while the operator is still typing the password
- a plain passphrase emits a startup warning recommending encryption

Payout and notification safety:

- a payout that falls due while locked is deferred, not lost: pending rewards stay in the database and are processed immediately on unlock, without waiting for the next cron tick
- ADM notifications require signing, so they are paused while locked; Slack keeps working and remains the reliable channel for lock alerts

`adm-pool` control CLI and `/api/health`:

- `adm-pool encrypt | unlock | lock | status` manages encryption and runtime lock state, talking to the running pool over a local Unix socket with owner-only (`0600`) permissions
- `unlock` checks the pool is reachable before prompting, so a stopped pool is reported immediately instead of after the password is typed
- `GET /api/health` returns a secret-free JSON snapshot for monitoring such as Zabbix, and `/api/config` exposes a `locked` flag for the dashboard

Frontend, docs, and tests:

- the dashboard shows `(waiting for password to unlock payouts…)` next to the delegate name while the pool is locked
- fixes deprecated/removed TypeScript options in `web/jsconfig.json` (`importsNotUsedAsValues` → `verbatimModuleSyntax`, `moduleResolution` → `Bundler`, drops unused `baseUrl`)
- documents the workflow in `README.md` and `config.default.jsonc`, and adds round-trip, wrong-password, tamper, and detection tests for the crypto helper

## Related issue

Closes #12

## External links (optional)

- Issue #12: https://github.com/Adamant-im/pool/issues/12
- Upstream node issue: https://github.com/Adamant-im/adamant/issues/247

## Screenshots or videos (optional)

Dashboard while locked, next to the delegate name:

```text
Delegate noisy_pool (waiting for password to unlock payouts…) distributes 80% rewards to voters …
```

## Breaking changes

No intentional breaking changes.

- A plain `passPhrase` continues to work with no migration. Encryption is opt-in.
- A new optional `controlSocket` config field is added; when empty, the socket path is derived from the pool port.
- `/api/config` gains an additive `locked` boolean. No existing field is removed or changed.

## How to test

```sh
npm --prefix server run lint
npm --prefix server test
npm --prefix web run lint
npm --prefix web run build
git diff --check
```

Manual control flow:

```sh
npm run adm-pool encrypt        # paste the admpool-enc-v1.… value into config as passPhrase
npm start                       # in a terminal: prompts for the operator password
# or, under pm2 (no TTY): starts LOCKED, then from another shell:
npm run adm-pool status
npm run adm-pool unlock
npm run adm-pool lock
curl http://localhost:36667/api/health
```

Expected result: all validation commands pass. Local run: server lint clean and 69 tests passing across 9 suites; web lint clean and `vite build` succeeds.

## Notes for reviewers

Secret safety:

- the operator password and decrypted passphrase exist only in process memory; only the public key and address are embedded in the encrypted blob, and both are already public on-chain
- the control socket is local-only with `0600` permissions and accepts no network unlock; no secret is logged at any level
- a wrong password or tampered blob is rejected by GCM authentication and by re-deriving and comparing the public key

Payout safety:

- the only signing paths (`pay_out.js`, `notify.js`) now read the passphrase through the secret store and no-op while locked; a deferred scheduled payout is replayed once, immediately on unlock
- payout accounting (`pending - FEE`, `received`, retries) is unchanged; the lock only gates whether signing is allowed

Compatibility and reliability:

- plain passphrase detection and the existing keypair derivation path are unchanged
- node failover and locked-state handling keep the pool serving read-only data while waiting for unlock

## Checklist

- [x] Code is formatted
- [x] Tests added/updated (if needed)
- [x] Documentation updated (if needed)
- [x] Changes tested in all relevant environments

### CODE REVIEW

File: server/src/helpers/control/socketPath.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * host do not collide.
+ * @param {object} config Loaded pool configuration
+ * @returns {string} Absolute path to the control socket
+ */
+export function resolveControlSocketPath(config) {
+  if (config.controlSocket) {
+    return config.controlSocket;
+  }
+
+  return path.join(os.tmpdir(), `adamant-pool-${config.port}-control.sock`);
```
Comment (dev-adamant-im) [inline server/src/helpers/control/socketPath.js]: Please avoid deriving the default control socket directly in `os.tmpdir()` with a predictable name. On multi-user hosts `/tmp` is world-writable, so another local user can pre-create this path and make `server.listen()` fail, leaving an encrypted pm2/systemd pool locked with no CLI unlock path; there is also a small permission window before `chmod(0600)` in `startControlServer()`. The issue calls for a local owner-only control socket, so the default should live under a private `0700` runtime directory (or require `controlSocket` to point to one), and startup should fail closed if the socket cannot be protected.

File: server/src/cron/payout.cron.js [hunk 2]
```diff
@@ -55,5 +59,36 @@ export default {
...
+   * Processes a payout that was due while the pool was locked. No-op when no
+   * scheduled run was missed.
+   * @returns {Promise<void>}
+   */
+  async runDeferred() {
+    if (!this.deferredPayout) {
+      return;
+    }
+
+    this.deferredPayout = false;
```
Comment (dev-adamant-im) [inline server/src/cron/payout.cron.js]: Please keep `deferredPayout` set until the deferred payout attempt actually runs while unlocked. Right now the flag is cleared before `payer.payOut()`. If the operator re-locks immediately after unlock, or if `payer.payOut()` throws before scheduling its normal retry, the missed scheduled payout is forgotten and a later unlock will not replay it; processing then waits for the next cron tick. Clear the flag only after a successful unlocked attempt, or restore it in the locked/error path.

**Comment by dev-adamant-im** [comment]
Review summary for PR #13:

What looks good:

- The task is valid and worth doing: encrypting the pool passphrase at rest reduces the impact of a leaked local config, and `/api/health` gives operators a clean monitoring surface without scraping logs.
- The PR broadly implements issue #12: encrypted/passphrase detection is deterministic, the signing paths now go through the runtime secret store, locked mode preserves pending rewards, ADM notifications are gated while Slack still works, and the dashboard/API expose lock state without returning the passphrase.
- I did not find consensus-breaking changes. The patch does not alter block validation, reward calculation inputs, vote accounting, node failover, or blockchain transaction format; it only gates outgoing signed actions while locked.

What needs attention:

- I left inline comments for two issues:
  - The default control socket path is predictable under the system temp directory, which weakens the local-only owner-controlled socket guarantee and can block CLI unlock on multi-user hosts.
  - `runDeferred()` clears the deferred payout marker before the payout attempt actually succeeds, so a re-lock or unexpected payout error can lose the immediate catch-up trigger.
- Please also add focused tests around the deferred payout marker and, ideally, the locked/unlocked control-flow edge cases. The new crypto helper has useful round-trip/tamper tests, but the payout catch-up behavior is the riskier operator-facing invariant here.

Validation I ran:

```sh
git diff --check origin/dev...HEAD
npm --prefix server run lint
npm --prefix server test
npm --prefix web run lint
npm --prefix web run build
npm install --package-lock-only --ignore-scripts --dry-run
npm --prefix server install --package-lock-only --ignore-scripts --dry-run
npm --prefix web install --package-lock-only --ignore-scripts --dry-run
printf '...test input...' | npm run adm-pool -- encrypt
npm run adm-pool -- status
curl -fsS http://localhost:36667/api/health
```

Results: validation passed. Server tests passed (69 tests across 9 suites), with the existing Jest warning that a worker was force-exited/open handles may remain. The health response I checked was secret-free and reported lock/passphrase state as expected.

**Review by dev-adamant-im — APPROVED** [review]
In advance 👍

File: server/src/helpers/control/socketPath.js (hunk 1 above)
Comment (massivedev0) [reply server/src/helpers/control/socketPath.js]: Fixed. The default control socket no longer lives directly in `os.tmpdir()`. It is now placed in a private per-user runtime directory `adamant-pool-<uid>` that `startControlServer()` creates with mode `0700` and **fails closed** on if the directory already exists but is owned by another user — closing the predictable-precreate vector on multi-user hosts. The socket inside it is chmod `0600`, and if that chmod fails the server now shuts the socket down (fail closed) rather than just warning; `app.js` exits with a fatal log if the control socket cannot be started securely. A custom `controlSocket` is still honored as the operator chose it.

Verified locally: dir is `drwx------ <uid>` and socket `srw------- <uid>`.

Files: `server/src/helpers/control/socketPath.js`, `server/src/api/control.js`, `server/src/app.js`. Not pushed yet — left uncommitted for local review.

File: server/src/cron/payout.cron.js (hunk 2 above)
Comment (massivedev0) [reply server/src/cron/payout.cron.js]: Fixed. `runDeferred()` now clears `deferredPayout` **only after** a payout attempt actually runs while unlocked. It no-ops and keeps the marker if the pool is locked again (`!secret.isUnlocked()`), and if `payer.payOut()` throws the marker is kept so the next unlock replays the missed run.

Also added focused tests for the marker across the edge cases you flagged — locked/unlocked scheduled runs, re-lock before catch-up, successful unlocked attempt, and a throwing attempt: `server/tests/helpers/cron_deferred.test.js` (6 tests). Full server suite is now 75 tests across 10 suites. Not pushed yet — left uncommitted for local review.

**Comment by massivedev0** [comment]
Thanks for the review. Both inline findings are addressed (replies in-thread):

- **Control socket path** — default socket moved into a private per-user `0700` runtime directory, fails closed on foreign ownership / unprotectable socket; custom `controlSocket` still honored.
- **Deferred payout marker** — `runDeferred()` clears the marker only after a real unlocked attempt; re-lock or a payout error keeps it pending for the next unlock.

On your testing note: I added `server/tests/helpers/cron_deferred.test.js` covering the deferred-payout invariant (locked/unlocked scheduled runs, re-lock before catch-up, success clears the marker, throw keeps it). Server suite is now 75 tests across 10 suites.

Validation run locally: `npm --prefix server run lint` clean, `npm --prefix server test` 75/75, `npm --prefix web run lint` clean, `npm --prefix web run build` succeeds. Verified the socket dir is `drwx------` and the socket `srw-------`.

Changes are not pushed yet (kept uncommitted for a local review pass); will push after.

**Comment by dev-adamant-im** [comment]
Re-review after the latest fixes:

What looks good:

- The previous control-socket finding is addressed in the current head. The default socket now resolves under a per-user runtime directory, the server prepares that directory with owner-only permissions, rejects foreign ownership, and keeps the socket itself at `0600`.
- The previous deferred-payout finding is addressed. `runDeferred()` now keeps the marker while the pool is locked or when the payout attempt throws, and clears it only after an unlocked attempt runs.
- The new `cron_deferred` tests cover the important catch-up cases: locked scheduled runs, unlocked scheduled runs, re-lock before catch-up, successful catch-up, and thrown payout attempts.
- I did not find new consensus, payout-accounting, secret-exposure, or health-endpoint regressions in the updated diff.

Validation I re-ran on head `608b8f37aa40d4f3a8ef1ad1e2f66c60696618f3`:

```sh
git diff --check origin/dev...HEAD
npm --prefix server run lint
npm --prefix server test
npm --prefix web run lint
npm --prefix web run build
npm install --package-lock-only --ignore-scripts --dry-run
npm --prefix server install --package-lock-only --ignore-scripts --dry-run
npm --prefix web install --package-lock-only --ignore-scripts --dry-run
printf '...test input...' | npm run adm-pool -- encrypt
```

Results: all commands passed. Server tests now pass with 75 tests across 10 suites. The Jest open-handle/force-exit warning is still present, same as before, but the suite exits successfully.

No new inline findings from this pass.
